### PR TITLE
[3.4] Additional and cleaned-up logging

### PR DESCRIPTION
**Pull request by danielhlarkin**

### CODE REVIEW

**Comment by danielhlarkin** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/1315/

File: arangod/IResearch/IResearchLink.cpp [hunk 1]
```diff
@@ -498,11 +500,10 @@ int IResearchLink::unload() {
...
     auto res = drop();
 
-    if (TRI_ERROR_NO_ERROR != res) {
-      LOG_TOPIC(WARN, arangodb::iresearch::TOPIC)
-        << "failed to drop collection from view while unloading dropped IResearch link '" << _id
-        << "' for IResearch view '" << _view->id() << "'";
-    }
+    LOG_TOPIC_IF(WARN, arangodb::iresearch::TOPIC, TRI_ERROR_NO_ERROR != res)
+        << "failed to drop collection from view while unloading dropped "
+        << "IResearch link '" << _id << "' for IResearch view '"
```
Comment (jsteemann) [inline arangod/IResearch/IResearchLink.cpp]: should be "arangosearch"

File: arangod/Cluster/TraverserEngine.cpp [hunk 2]
```diff
@@ -201,11 +201,12 @@ bool BaseEngine::lockCollection(std::string const& shard) {
...
   Result lockResult = _trx->lockRecursive(cid, AccessMode::Type::READ);
-  
+
   if (!lockResult.ok() && !lockResult.is(TRI_ERROR_LOCKED)) {
-    LOG_TOPIC(ERR, arangodb::Logger::FIXME)
+    LOG_TOPIC(ERR, arangodb::Logger::CLUSTER)
         << "Locking shard " << shard << " lead to exception '"
-        << lockResult.errorNumber() << "' (" << lockResult.errorMessage() << ") ";
+        << lockResult.errorNumber() << "' ("
+        << lockResult.errorMessage() << ") ";
```
Comment (jsteemann) [inline arangod/Cluster/TraverserEngine.cpp]: can remove trailing whitespace in message

File: arangod/IResearch/IResearchView.cpp [hunk 3]
```diff
@@ -1409,6 +1410,9 @@ arangodb::Result IResearchView::commit() {
     return {}; // nothing more to do
   }
 
+  LOG_TOPIC(TRACE, arangodb::iresearch::TOPIC)
+      << "begining flush commit for " << name();
```
Comment (jsteemann) [inline arangod/IResearch/IResearchView.cpp]: beginning (double n)

File: arangod/IResearch/IResearchView.cpp [hunk 4]
```diff
@@ -1925,6 +1937,8 @@ PrimaryKeyIndexReader* IResearchView::snapshot(
   TRI_ASSERT(reader);
 
   if (!_asyncSelf->get()) {
+    LOG_TOPIC(WARN, arangodb::iresearch::TOPIC)
+      << "current view no longer valid while creating IResearchView snapshot";
```
Comment (jsteemann) [inline arangod/IResearch/IResearchView.cpp]: arangosearch snapshot

File: arangod/IResearch/IResearchViewCoordinator.cpp [hunk 5]
```diff
@@ -86,20 +88,31 @@ arangodb::Result createLink(
   );
 
   if (!mergeSliceSkipKeys(builder, link, acceptor)) {
+    LOG_TOPIC(DEBUG, arangodb::iresearch::TOPIC)
+        << "failed to update link definition with the view name while updating "
+        << "IResearch view '" << view.name() << "' and collection '"
```
Comment (jsteemann) [inline arangod/IResearch/IResearchViewCoordinator.cpp]: "arangosearch view"

File: arangod/RocksDBEngine/RocksDBEngine.cpp [hunk 6]
```diff
@@ -1845,6 +1880,9 @@ std::unique_ptr<TRI_vocbase_t> RocksDBEngine::openExistingDatabase(
   auto vocbase =
       std::make_unique<TRI_vocbase_t>(TRI_VOCBASE_TYPE_NORMAL, id, name);
 
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+  //std::vector<iresearch::IResearchView*> views;
```
Comment (jsteemann) [inline arangod/RocksDBEngine/RocksDBEngine.cpp]: can remove this entirely I think.

File: arangod/RocksDBEngine/RocksDBEngine.cpp [hunk 7]
```diff
@@ -1911,17 +1957,26 @@ std::unique_ptr<TRI_vocbase_t> RocksDBEngine::openExistingDatabase(
...
       physical->deserializeKeyGenerator(settingsManager());
-      LOG_TOPIC(DEBUG, arangodb::Logger::FIXME) << "added document collection '"
-                                                << collection->name() << "'";
+      LOG_TOPIC(DEBUG, arangodb::Logger::ENGINES) << "added document collection '"
+                                                << collection->name() << "' with properties " << it.toJson();
     }
 
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+/*    for (auto view : views) {
+      TRI_ASSERT(view != nullptr);
```
Comment (jsteemann) [inline arangod/RocksDBEngine/RocksDBEngine.cpp]: remove?

File: arangod/RocksDBEngine/RocksDBKey.h [hunk 8]
```diff
@@ -282,7 +282,7 @@ class RocksDBKey {
     TRI_ASSERT(size >= sizeof(char));
 
     const auto type = static_cast<RocksDBEntryType>(data[0]);
-#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+/*#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
```
Comment (jsteemann) [inline arangod/RocksDBEngine/RocksDBKey.h]: is this intentionally commented out? does this change the behavior somehow?

**Pull request merged**